src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,573 @@
+        {
+            ReturnErrorOnFailure(reader.Get(nameSpan));
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(TagScene::kTransitionTimeMs)));
+        }
+        // Empty name will be initialized if the name wasn't stored
+        mStorageData.SetName(nameSpan);
+
+        ReturnErrorOnFailure(reader.Get(mStorageData.mSceneTransitionTimeMs));
+        ReturnErrorOnFailure(
+            mStorageData.mExtensionFieldSets.Deserialize(reader, TLV::ContextTag(TagScene::kExtensionFieldSetsContainer)));
+
+        return reader.ExitContainer(container);
+    }
+};
+
+// A Full fabric serialized TLV length is 88 bytes, 128 bytes gives some slack.  Tested by running writer.GetLengthWritten at the
+// end of the Serialize method of FabricSceneData
+static constexpr size_t kPersistentFabricBufferMax = 128;
+
+/**
+ * @brief Linked list of all scenes in a fabric, stored in persistent memory
+ *
+ * FabricSceneData is an access to a linked list of scenes
+ */
+struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
+{
+    FabricIndex fabric_index = kUndefinedFabricIndex;
+    uint8_t scene_count      = 0;
+    SceneStorageId scene_map[kMaxScenesPerFabric];
+
+    FabricSceneData() = default;
+    FabricSceneData(FabricIndex fabric) : fabric_index(fabric) {}
+
+    CHIP_ERROR UpdateKey(StorageKeyName & key) override
+    {
+        VerifyOrReturnError(kUndefinedFabricIndex != fabric_index, CHIP_ERROR_INVALID_FABRIC_INDEX);
+        key = DefaultStorageKeyAllocator::FabricSceneDataKey(fabric_index);
+        return CHIP_NO_ERROR;
+    }
+
+    void Clear() override
+    {
+        scene_count = 0;
+        for (uint8_t i = 0; i < kMaxScenesPerFabric; i++)
+        {
+            scene_map[i].Clear();
+        }
+    }
+
+    CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override
+    {
+        TLV::TLVType container;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, container));
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagScene::kSceneCount), scene_count));
+        ReturnErrorOnFailure(writer.StartContainer(TLV::ContextTag(TagScene::kStorageIDArray), TLV::kTLVType_Array, container));
+
+        // Storing the scene map
+        for (uint8_t i = 0; i < kMaxScenesPerFabric; i++)
+        {
+            ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, container));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagScene::kEndpointID), (scene_map[i].mEndpointId)));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagScene::kGroupID), (scene_map[i].mGroupId)));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(TagScene::kSceneID), (scene_map[i].mSceneId)));
+            ReturnErrorOnFailure(writer.EndContainer(container));
+        }
+        ReturnErrorOnFailure(writer.EndContainer(container));
+        return writer.EndContainer(container);
```

Comment:
Using "container" for all three of those EndContainer calls is likely not going to work right.  There should be three separate variables, and the one used for StartContainer should be used for the corresponding EndContainer.